definition_validator/validator.cc

```diff
@@ -331,7 +345,8 @@ void validateOverriding(const core::Context ctx, core::SymbolRef method) {
                 e.addErrorLine(overridenMethod.data(ctx)->loc(), "defined here");
             }
         }
-        if ((overridenMethod.data(ctx)->isAbstract() || overridenMethod.data(ctx)->isOverridable()) &&
+        if ((overridenMethod.data(ctx)->isAbstract() || overridenMethod.data(ctx)->isOverridable() ||
+             (overridenMethod.data(ctx)->hasSig() && method.data(ctx)->isOverride())) &&
```

Comment:
This is the important part of this PR: we call `validateCompatibleOverride` when the overridden method has a signature, and the current method is an override.